vault/resource_ad_secret_backend.go

```diff
@@ -0,0 +1,645 @@
+			Optional:    true,
+			Description: `Use anonymous bind to discover the bind DN of a user (optional)`,
+		},
+		"formatter": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Computed:    true,
+			Deprecated:  `Formatter is deprecated and password_policy should be used.`,
+			Description: `Text to insert the password into, ex. "customPrefix{{PASSWORD}}customSuffix".`,
+		},
+		"groupattr": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: `LDAP attribute to follow on objects returned by <groupfilter> in order to enumerate user group membership. Examples: "cn" or "memberOf", etc. Default: cn`,
+		},
+		"groupdn": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: `LDAP search base to use for group membership search (eg: ou=Groups,dc=example,dc=org)`,
+		},
+		"groupfilter": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: `Go template for querying group membership of user (optional) The template can access the following context variables: UserDN, Username Example: (&(objectClass=group)(member:1.2.840.113556.1.4.1941:={{.UserDN}})) Default: (|(memberUid={{.Username}})(member={{.UserDN}})(uniqueMember={{.UserDN}}))`,
+		},
+		"insecure_tls": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Description: `Skip LDAP server SSL Certificate verification - VERY insecure (optional)`,
+		},
+		"last_rotation_tolerance": {
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Computed:    true,
+			Description: `The number of seconds after a Vault rotation where, if Active Directory shows a later rotation, it should be considered out-of-band.`,
+		},
+		"length": {
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Computed:    true,
+			Deprecated:  `Length is deprecated and password_policy should be used.`,
+			Description: `The desired length of passwords that Vault generates.`,
+		},
+		"local": {
+			Type:        schema.TypeBool,
+			Required:    false,
+			Optional:    true,
+			Description: "Mark the secrets engine as local-only. Local engines are not replicated or removed by replication.Tolerance duration to use when checking the last rotation time.",
+		},
+		"max_lease_ttl_seconds": {
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Computed:    true,
+			Description: "Maximum possible lease duration for secrets in seconds",
+		},
+		"max_ttl": {
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Computed:    true,
+			Description: `In seconds, the maximum password time-to-live.`,
+		},
+		"password_policy": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: `Name of the password policy to use to generate passwords.`,
+		},
+		"request_timeout": {
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Description: `Timeout, in seconds, for the connection when making requests against the server before returning back an error.`,
+		},
+		"starttls": {
```

Comment:
How come there's no underscore in this one?

Reply:
This configurable comes from Vault's LDAP client so we don't have a say here.